packages/components/src/tabs/index.tsx

```diff
@@ -0,0 +1,188 @@
+/**
+ * External dependencies
+ */
+import * as Ariakit from '@ariakit/react';
+import cx from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useLayoutEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import type { TabPanelProps } from '../tab-panel/types';
+import { usePrevious } from '@wordpress/compose';
+
+/**
+ * Tabs is an ARIA-compliant tabpanel.
+ *
+ * Tabs organizes content across different screens, data sets, and interactions.
+ * It has two sections: a list of tabs, and the view to show when tabs are chosen.
+ *
+ * ```jsx
+ * import { Tabs } from '@wordpress/components';
+ *
+ * const onSelect = ( tabName ) => {
+ *   console.log( 'Selecting tab', tabName );
+ * };
+ *
+ * const MyTabs = () => (
+ *   <Tabs
+ *     className="my-tabs"
+ *     activeClass="active-tab"
+ *     onSelect={ onSelect }
+ *     tabs={ [
+ *       {
+ *         name: 'tab1',
+ *         title: 'Tab 1',
+ *         className: 'tab-one',
+ *       },
+ *       {
+ *         name: 'tab2',
+ *         title: 'Tab 2',
+ *         className: 'tab-two',
+ *       },
+ *     ] }
+ *   >
+ *     { ( tab ) => <p>{ tab.title }</p> }
+ *   </Tabs>
+ * );
+ * ```
+ */
+
+export const Tabs = ( props: TabPanelProps ) => {
+	const {
+		tabs,
+		children,
+		onSelect,
+		className,
+		orientation = 'horizontal',
+		selectOnMove = true,
+		initialTabName,
+		activeClass = 'is-active',
+	} = props;
+
+	const tabStore = Ariakit.useTabStore( {
+		setSelectedId: ( newTabValue ) => {
+			if ( typeof newTabValue === 'undefined' || newTabValue === null ) {
+				return;
+			}
+
+			const newTab = tabs.find( ( t ) => t.name === newTabValue );
+			if ( newTab?.disabled || newTab === selectedTab ) {
+				return;
+			}
+
+			onSelect?.( newTabValue );
+		},
+		orientation,
+		selectOnMove,
+		defaultSelectedId: initialTabName,
```

Comment:
I wonder if this would make sense?

```suggestion
		defaultSelectedId: initialTabName ?? tabs[ 0 ]?.name,
```

Reply:
Initially I didn't think this would change the behavior too much, with both Ariakit and our own custom logic autoselected the first available tab, but it actually did. Currently, the existing `TabPanel` component will call the `onSelect` callback (if provided) on the initial render. The above change would negate that.

🤔 Whether or not that initial `onSelect` call is truly desirable or if it's just "how things are" might be an interesting question, but for now I'm leaning towards leaving it as-is to maintain parity with the existing component. Definitely interested your thoughts on this part though @mirka

Reply:
> for now I'm leaning towards leaving it as-is to maintain parity with the existing component

I agree that we should maintain the `onSelect` behavior in the existing component for parity. But I'm pretty sure it's undesirable, so we should fix it in the v2 component. `onSelect` should only be called for user-initiated changes.